lib/__tests__/index.test.ts

```diff
@@ -184,3 +185,123 @@ describe('Mock Console Log', () => {
         mockLog.mockRestore();
     });
 });
+
+describe('mockedRun', () => {
+    let mockRun: (_: () => any) => MockedRunResult;
+    let mocks: MockedRunResult;
+
+    beforeAll(() => {
+        mockRun = mockedRun({
+            stdout: mockProcessStdout,
+            stderr: mockProcessStderr,
+            exit: mockProcessExit,
+            log: mockConsoleLog,
+        });
+    });
+
+    it('should call every mock once', () => {
+        mocks = mockRun(() => {
+            process.stdout.write('stdout payload');
+            process.stderr.write('stderr payload');
+            process.exit(-1);
+            console.log('log payload');
+        });
+        expect(mocks.stdout).toHaveBeenCalledTimes(1);
+        expect(mocks.stderr).toHaveBeenCalledTimes(1);
+        expect(mocks.exit).toHaveBeenCalledTimes(1);
+        expect(mocks.log).toHaveBeenCalledTimes(1);
+    });
+
+    it('should receive the correct arguments', () => {
+        mocks = mockRun(() => {
+            process.stdout.write('stdout payload');
+            process.stderr.write('stderr payload');
+            process.exit(-1);
+            console.log('log payload');
+        });
+        expect(mocks.stdout).toHaveBeenCalledWith('stdout payload');
+        expect(mocks.stderr).toHaveBeenCalledWith('stderr payload');
+        expect(mocks.exit).toHaveBeenCalledWith(-1);
+        expect(mocks.log).toHaveBeenCalledWith('log payload');
+    });
+
+    it('should receive the correct return value', () => {
+        mocks = mockRun(() => {
+            return 'return string';
+        });
+        expect(mocks.result).toEqual('return string');
+    });
+
+    it('should receive the correct thrown value', () => {
+        const expectedError = new Error('return string');
+        mocks = mockRun(() => {
+            throw expectedError;
+        });
+        expect(mocks.error).toEqual(expectedError);
+    });
+});
+
+describe('asyncMockedRun', () => {
+    let mockRun: (_: () => any) => Promise<MockedRunResult>;
+    let mocks: MockedRunResult;
+
+    beforeAll(() => {
+        mockRun = asyncMockedRun({
```

Comment:
Any reason to have this inside a `beforeAll`? I think we could just assign this on `describe`'s body, no?

Reply:
No problem.

Reply:
Seems good to me.